ckan/lib/search/query.py

```diff
@@ -89,29 +92,48 @@ def convert_legacy_parameters_to_solr(
     return solr_params
 
 
+def _parse_local_params(local_params: str) -> list[Union[str, list[str]]]:
+    """
+    Parse a local parameters section as return it as a list, eg:
+
+    {!dismax qf=myfield v='some value'} -> ['dismax', ['qf', 'myfield'], ['v', 'some value']]
+
+
+    {!type=dismax qf=myfield v='some value'} -> [['type', 'dismax'], ['qf', 'myfield'], ['v', 'some value']]
+
+    """
+    key = Word(alphas)
+    value = QuotedString('"') | QuotedString("'") | Word(alphanums)
+    pair = Group(key + Suppress("=") + value)
+    expression = Suppress("{!") + OneOrMore(pair | key) + Suppress("}")
+
+    return expression.parse_string(local_params).as_list()
+
+
 def _get_local_query_parser(q: str) -> str:
     """
-    Given a Solr `q` parameter, extract any custom query parsers used in the
+    Given a Solr parameter, extract any custom query parsers used in the
     local parameters, .e.g. q={!child ...}...
     """
     qp_type = ""
     q = q.strip()
     if not q.startswith("{!"):
         return qp_type
 
-    local_params = q[:q.index("}")]
+    local_params = q[:q.index("}") + 1]
```

Comment:
in case there's a `}` inside a quoted string in the local params
```suggestion
    local_params = q[:q.rindex("}") + 1]
```

Reply:
Also a possible ValueError when missing